src/service/chat.js

```diff
@@ -206,6 +206,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatFromPost` function seems to be handling errors correctly and returning promises as expected. However, the payload construction could be simplified by using a ternary operator instead of a switch statement, since there are only two cases. This would make the code more concise and easier to read.
```diff
-  let payload = {source};
-  switch (source) {
-    case 'post':
-      payload = {...payload, postId: id};
-      break;
-    case 'comment':
-      payload = {...payload, commentId: id};
-      break;
-    default:
-      break;
-  }
+  const payload = {
+    source,
+    ...(source === 'post' ? {postId: id} : {commentId: id})
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->